packages/fluentui/docs/src/prototypes/table/SelectableTable.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  gridCellWithFocusableElementBehavior,
+  gridNestedBehavior,
+  gridCellBehavior,
+  gridRowBehavior,
+} from '@fluentui/accessibility';
+import { Table, Checkbox, Box } from '@fluentui/react-northstar';
+import * as React from 'react';
+
+interface SelectableTableAction {
+  type: 'TOGGLE_ITEM' | 'TOGGLE_ALL';
+  checked: boolean;
+  itemIndex?: number;
+}
+
+interface SelectableTableState {
+  allRowsSelected: boolean;
+  selectedRows: boolean[];
+}
+
+const SelectableTableStateReducer = (state: SelectableTableState, action: SelectableTableAction) => {
+  switch (action.type) {
+    case 'TOGGLE_ITEM': {
+      const selectedRows = state.selectedRows.slice(0);
```

Comment:
```suggestion
      const selectedRows = [...state.selectedRows];
```

_nit_ Feel free to ignore